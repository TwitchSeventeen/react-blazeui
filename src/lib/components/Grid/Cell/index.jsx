import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Cell(props) {
  const {
    align,
    children,
    className,
    offset,
    width,
    xsmall,
    small,
    medium,
    large,
    xlarge,
    huge,
    noGutter,
    fixedWidth,
    dataName,
    ...other
  } = props;
  const classes = classNames(
    'o-grid__cell',
    { [`o-grid__cell--width-${width}`]: width },
    { [`o-grid__cell--offset-${offset}`]: offset },
    { [`o-grid__cell--${align}`]: align },
    { 'o-grid__cell--no-gutter': noGutter },
    { 'o-grid__cell--width-fixed': fixedWidth },
    { [`o-grid__cell--width-${xsmall}@xsmall`]: xsmall },
    { [`o-grid__cell--width-${small}@small`]: small },
    { [`o-grid__cell--width-${medium}@medium`]: medium },
    { [`o-grid__cell--width-${large}@large`]: large },
    { [`o-grid__cell--width-${xlarge}@xlarge`]: xlarge },
    { [`o-grid__cell--width-${huge}@super`]: huge },
    className,
  );
  return (
    <div
      {...other}
      className={classes}
      data-name={dataName}
    >
      { children }
    </div>
  );
}

Cell.propTypes = {
  /**
   * Vertical alignment of the Cell
   */
  align: PropTypes.oneOf([
    'top',
    'center',
    'bottom',
  ]),
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * The offset of this cell as a percentage of its parent.
   * Must be a multiple of 5, or 33, 66.
   */
  offset: PropTypes.number,
  /**
   * Width of this cell as a percentage of its parent.
   * Must be a multiple of 5, or 33, 66.
   */
  width: PropTypes.number,
  /**
   * Cell display for xsmall screens
   * Must be a multiple of 5, or 33, 66.
   */
  xsmall: PropTypes.number,
  /**
   * Cell display for small screens
   * Must be a multiple of 5, or 33, 66.
   */
  small: PropTypes.number,
  /**
   * Cell display for medium screens
   * Must be a multiple of 5, or 33, 66.
   */
  medium: PropTypes.number,
  /**
   * Cell display for large screens
   * Must be a multiple of 5, or 33, 66.
   */
  large: PropTypes.number,
  /**
   * Cell display for xlarge screens
   * Must be a multiple of 5, or 33, 66.
   */
  xlarge: PropTypes.number,
  /**
   * Cell display for super screens
   * Must be a multiple of 5, or 33, 66.
   */
  huge: PropTypes.number,
  /**
   * Specifies no spacing for the Cell
   */
  noGutter: PropTypes.bool,
  /**
   * Specifies this Cell has a fixed width
   */
  fixedWidth: PropTypes.bool,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
};

Cell.defaultProps = {
  align: null,
  children: null,
  className: null,
  offset: null,
  width: null,
  xsmall: null,
  small: null,
  medium: null,
  large: null,
  xlarge: null,
  huge: null,
  noGutter: false,
  fixedWidth: false,
  dataName: 'grid-cell',
};

export default Cell;
