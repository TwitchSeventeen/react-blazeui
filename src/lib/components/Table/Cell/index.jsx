import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Cell(props) {
  const {
    children, className, ...other
  } = props;

  const classes = classNames(
    'c-table__cell',
    className,
  );

  return (
    <td className={classes} {...other}>
      { children }
    </td>
  );
}

Cell.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
};

Cell.defaultProps = {
  className: null,
  children: null,
};

export default Cell;
