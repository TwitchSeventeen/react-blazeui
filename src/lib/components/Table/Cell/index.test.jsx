import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Cell from './index';

describe('Cell', () => {
  it('should be defined', () => {
    expect(Cell).toBeDefined();
  });

  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <Cell>
        lorem ipsum dolor
      </Cell>,
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
