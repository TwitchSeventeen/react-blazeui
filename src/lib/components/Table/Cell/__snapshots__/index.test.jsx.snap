// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell renders correctly 1`] = `
<td
  className="c-table__cell"
>
  lorem ipsum dolor
</td>
`;
